gulp3.babel.js

```diff
@@ -0,0 +1,164 @@
+        'browser-sync'
+      ],
+      rename: {
+        'gulp-util': 'gutil',
+        'run-sequence': 'sequence',
+        'gulp-if': 'gulpIf'
+      }
+    });
+  }
+
+  const babelFn = require(
+    path.join(process.cwd(), 'gulp', 'tasks', 'babel')
+  );
+  const copyFn = require(
+    path.join(process.cwd(), 'gulp', 'tasks', 'copy')
+  );
+
+  const {eslint, sequence} = plugins;
+  const eslintConfig = makeEslintConfig({
+    basic: false,
+    react: true,
+    isDev: true,
+    lintEnv: 'build'
+  });
+  const eslintFn = () => {
+    return gulp.src(scripts)
+      .pipe(eslint(eslintConfig))
+      .pipe(eslint.format(formatter));
+  };
+  const config = {force, release};
+
+  gulp.task('babel', tasks.babel || babelFn(gulp, plugins, config));
+  gulp.task('lint', tasks.lint || eslintFn);
+  gulp.task('copy', tasks.copy || copyFn(gulp, plugins, config));
+  gulp.task('build', (cb) => {
+    sequence(
+      'lint',
+      ['copy', 'babel'],
+      cb
+    );
+  });
+
+  gulp.task('watch', ['lint', 'babel', 'copy'], () => {
+    gulp.watch(scripts, [], (cb) => {
+      sequence(
+        'lint',
+        ['babel'],
+        cb
+      );
+    });
+
+    gulp.watch('./packages/*/src/**/*.json', ['copy']);
+  });
+} else {
+  const build = require('./packages/boiler-core/src');
+  const {tasks, config, plugins: $} = build(gulp, {
+    //fp: 'boiler.custom.config.js'
+  });
+  const {sources, utils, environment, release} = config;
+  const {isDev} = environment;
+  const {testDir, buildDir} = sources;
+  const {addbase} = utils;
+
+  gulp.task('assemble', tasks.assemble);
+  gulp.task('babel', tasks.babel);
+  gulp.task('browser-sync', tasks.browserSync);
+  gulp.task('clean', tasks.clean);
+  gulp.task('copy', tasks.copy);
+  gulp.task('karma', tasks.karma);
+  gulp.task('lint:test', tasks.eslint);
+  gulp.task('lint:build', tasks.eslint);
+  gulp.task('lint', ['lint:test', 'lint:build']);
+  gulp.task('mocha', tasks.mocha);
+  gulp.task('nodemon', tasks.nodemon);
+  gulp.task('selenium', tasks.selenium);
+  gulp.task('selenium:tunnel', tasks.selenium);
+  gulp.task('webpack:global', tasks.webpack);
+  gulp.task('webpack:main', tasks.webpack);
+  gulp.task('webpack:server', tasks.webpack);
+  gulp.task('webpack', ['webpack:global', 'webpack:main']);
+
+  gulp.task('build', (cb) => {
+    if (isDev) {
+      //gulp watch
+      $.sequence(
+        'clean',
+        'copy',
+        'lint',
+        'webpack',
+        'assemble',
+        'browser-sync',
+        cb
+      );
+    } else if (release) {
+      $.sequence(
+        'clean',
+        'copy',
+        'lint',
+        'babel',
+        cb
+      );
+    } else {
+      $.sequence(
+        'clean',
+        'copy',
+        'lint',
+        'webpack',
+        'webpack:server',
+        'assemble',
+        cb
+      );
+    }
+  });
+
+  gulp.task('default', ['build']);
+
+
+  gulp.task('test:integration', (cb) => {
+    $.sequence(
+      ['clean', 'lint'],
```

Comment:
This isn't relevant to this pull request, but is there any reason we invoke the `clean` task before running integration tests?  It seems like an unnecessary step for testing.